### PR TITLE
Migrate to eslint

### PR DESCRIPTION
**Pull request by fortuna**
I had to fix a ton of code to make this work.

Config was copied from https://github.com/lit/lit/blob/main/.eslintrc.json.
The config is for ES Modules, so I had to disable the require test in the Webpack files. We should migrate them to ESM at some point.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{
```
Comment (daniellacosse) [inline .eslintrc.json:1]: how much do we want to stick to lit's eslint config? I have a bunch of rules I'm happy to recommend

File: .eslintignore [hunk 2]
```diff
@@ -0,0 +1,2 @@
+/build/
+node_modules/
```
Comment (daniellacosse) [inline .eslintignore:2]: nit: newline

File: src/server_manager/web_app/app.ts [hunk 3]
```diff
@@ -437,7 +437,8 @@ export class App {
       // Wait for server config to load, then update the server view and list.
       if (isManagedServer(server)) {
         try {
-          for await (const _ of server.monitorInstallProgress()) {}
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          for await (const _ of server.monitorInstallProgress()) {/* empty */}
```
Comment (daniellacosse) [inline src/server_manager/web_app/app.ts:441]: i know it's not your code but wouldn't `await Promise.all(server.monitorInstallProgress())` work, here? I'm a bit confused. don't have to change haha

**Review by daniellacosse — APPROVED** [review]
👏

File: .eslintrc.json (hunk 1 above)
Comment (daniellacosse) [reply .eslintrc.json:1]: https://github.com/daniellacosse/only-web/blob/main/.eslintrc.cjs

Comment (fortuna) [reply .eslintrc.json:1]: I'd rather stick to Google's standards.

File: src/server_manager/web_app/app.ts (hunk 3 above)
Comment (fortuna) [reply src/server_manager/web_app/app.ts:441]: No, it's an async for loop with an async generator. You need to iterate.

Comment (daniellacosse) [reply src/server_manager/web_app/app.ts:441]: ah, it's a generator

File: package.json [hunk 4]
```diff
@@ -24,7 +26,7 @@
     "format:all": "prettier --write \"**/*.{cjs,html,js,json,md,ts}\"",
     "lint": "npm run lint:sh && npm run lint:ts",
     "lint:sh": "bash ./scripts/shellcheck.sh",
-    "lint:ts": "npx tslint 'src/**/*.ts' -e '**/node_modules/**'",
+    "lint:ts": "eslint \"**/*.{js,ts}\"",
```
Comment (jyyi1) [inline package.json:29]: Shall we include this as a step in CI pipeline as well? I mean the CI for every PR.

**Review by jyyi1 — APPROVED** [review]

File: package.json (hunk 4 above)
Comment (fortuna) [reply package.json:29]: `npm run lint` is already in the ci

**Pull request merged**